src/rpc/util.cpp

```diff
@@ -579,7 +579,7 @@ UniValue RPCHelpMan::HandleRequest(const JSONRPCRequest& request) const
     if (request.mode == JSONRPCRequest::GET_HELP || !IsValidNumArgs(request.params.size())) {
         throw std::runtime_error(ToString());
     }
-    const UniValue ret = m_fun(*this, request);
+    UniValue ret = m_fun(*this, request);
```

Comment:
I was trying to benchmark this, but couldn't see a difference. Steps to reproduce:


```
# start bitcoin core with "-rpcuser=user -rpcpassword=password -server=1 -rpcport=12345 -noconnect"
# create a reply with
python3 -c 'ff="ff"*2'000'000;print(f"{{\"jsonrpc\": \"1.0\", \"id\": \"curltest\", \"method\": \"echo\", \"params\": [\"{ff}\"]}}")' > /tmp/data.json
# Run ab
ab -p /tmp/data.json -n 100 -c 1 -A user:password "http://localhost:12345/"

Reply:
Running with `-n 10000 -c 4` I get mean time of 50ms on this branch, 51ms on master.

Reply:
I don't know if a +-1 difference is more than noise+rounding error when the resolution is quantized to 1ms.

Either the copy is so cheap that it doesn't matter in the pipe, or the compiler has already optimized away the `const`?

Reply:
Indeed, but there is clearly no regression and this PR is about introducing the new clang-tidy rule. It is not about any particular performance improvement but allowing us to automatically check for bad practices. Should we not bother trying to add new clang-tidy rules if there's no performance benefit?

Reply:
> this PR is about introducing the new clang-tidy rule. It is not about any particular performance improvement but allowing us to automatically check for bad practices.

Almost the same words I was going to put in my own comment :)

Reply:
clang, and gcc likely as well, seem to remove the `const` already on my machine. So any measurement difference is noise. For reference, on 20ff4991e548630d7bb5e491fa4d69ec49369872 I could see a 6% difference. So if we apply performance related checks, it might be good to prioritize the ones that make a difference. Otherwise we'll get people to change i++ to ++i for integers all over the source code.

Reply:
Sorry, but I do not realize how does not having a performance gain for a particular case with a small `UniValue` instance on a particular platform make forcing non-pessimized coding style questionable?

Reply:
It would be nice to be able to say: "This change has a positive effect on at least one end user or dev" instead of "This change doesn't change the binary and causes merge conflicts"